ugvc/mrd/featuremap_consensus_utils.py

```diff
@@ -0,0 +1,211 @@
+        end = int(genomic_interval_list[1].split("-")[1])
+
+    # generate a new header
+    orig_header = pysam.VariantFile(featuremap).header
+    header = pysam.VariantHeader()
+    header.add_meta("fileformat", "VCFv4.2")
+    # header.filters.add("PASS", None, None, "Aggregated quality above threshold, more than a single alternative read")
+    header.filters.add(
+        "SingleRead", None, None, "Aggregated quality above threshold, Only a single read agrees with alternative"
+    )
+    header.filters.add("LowQual", None, None, "Aggregated quality below threshold")
+    for field in fields_to_collect["numeric_aggregated_fields"]:
+        header.info.add(
+            field, orig_header.info[field].number, orig_header.info[field].type, orig_header.info[field].description
+        )
+    for field in fields_to_collect["numeric_array_fields"]:
+        header.info.add(field, ".", orig_header.info[field].type, orig_header.info[field].description)
+    for field in fields_to_collect["string_list_fields"]:
+        header.info.add(field, "1", orig_header.info[field].type, orig_header.info[field].description)
+    # exceptions: X_QUAL
+    header.info.add(FeatureMapFields.X_QUAL.value, ".", "Float", "Quality of reads containing this location")
+    header.formats.add(GT, 1, "String", "Genotype")
+    header.formats.add(DP, 1, "Integer", "Read depth")
+    header.formats.add(AD, "R", "Integer", "Allelic depths")
+    header.formats.add(VAF, 1, "Float", "Allele frequency")
+    # copy contigs from original header
+    for contig in orig_header.contigs:
+        header.contigs.add(contig)
+    header.add_sample(sample_name)
+
+    # open an output file
+    out_fh = pysam.VariantFile(output_vcf, "w", header=header)
+
+    cons_dict = defaultdict(dict)
+    with pysam.VariantFile(featuremap) as f:
+        prev_key = tuple()
+        for rec in f.fetch(chrom, start, end):
+            rec_id = (rec.chrom, rec.pos, rec.ref, rec.alts[0])
+            if FeatureMapFields.FILTERED_COUNT.value not in cons_dict[rec_id]:
+                if len(cons_dict.keys()) > 1:
+                    # write to file
+                    write_a_pileup_record(cons_dict[prev_key], prev_key, out_fh, header, min_qual)
+                    cons_dict.pop(prev_key)
+                # initialize the record
+                for field in fields_to_collect["numeric_aggregated_fields"]:
+                    cons_dict[rec_id][field] = 0
+                # exceptions: X_QUAL
+                cons_dict[rec_id]["X_QUAL"] = np.array([])
+                for field in fields_to_collect["numeric_array_fields"]:
+                    cons_dict[rec_id][field] = np.array([])
+                for field in fields_to_collect["string_list_fields"]:
+                    cons_dict[rec_id][field] = []
+            # update the record
+            for field in fields_to_collect["numeric_aggregated_fields"]:
+                cons_dict[rec_id][field] += 1
```

Comment:
Are "numeric_aggregated_fields" by definition counters or records? It looks like it to me, so in this case it should just be one variable rather than a list.

Reply:
it's a counter. Is there something I should change?

Reply:
I just meant that there is no reason to have a list of variables if they would all be counters, it can be just one variable